### PR TITLE
Alpha Animals update

### PR DESCRIPTION
**Pull request by feeddanoob**
## Additions
New creatures and one new projectile.

## References
- Contributes towards #580

## Reasoning
New animals needed to be balanced out for CE. That said I have no idea how to balance...So I decided to create the framework for whoever wants to balance them.

## Alternatives
Modifying alpha damages or armor penetration where needed

## Testing
Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [ ] Playtested a colony (specify how long)

### CODE REVIEW

**Comment by N7Huntsman** [comment]
The mantrap ranged attack currently throws errors.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrystallineCaracal.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+				<li Class="PatchOperationAdd">
+					<xpath>/Defs/ThingDef[defName="AA_CrystallineCaracal"]/statBases</xpath>
+					<value>
+						<MeleeDodgeChance>0.36</MeleeDodgeChance>
+						<MeleeCritChance>0.02</MeleeCritChance>
+						<MeleeParryChance>0.01</MeleeParryChance>
+					</value>
+				</li>
+				<li Class="PatchOperationReplace">
+					<xpath>/Defs/ThingDef[defName="AA_CrescendoAnole"]/tools</xpath>
```
Comment (N7Huntsman) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrystallineCaracal.xml:27]: This is missing sharp and blunt penetration.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
- These new creatures are missing the sharp and blunt penetration.
- The Mantrap projectile throws an error. Looks like the projectile hasn't been patched.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrystallineCaracal.xml (hunk 1 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrystallineCaracal.xml:27]: I knew I forgot something

**Comment by feeddanoob** [comment]
Okay. 
For the armor pen stuff do all creatures need them? Or should I guess by damage and/or animal looks? I added a patch to the aa_acidbolt (acid spit) and maybe that should clear something for that error... maybe (at least on startup it doesn't have errors).

**Comment by feeddanoob** [comment]
I did a quick test on the mantrap and no error log popped up for me so I think it is fixed

**Comment by randomtyper** [comment]
> For the armor pen stuff do all creatures need them? Or should I guess by damage and/or animal looks?
Every creature needs at least some, or they won't be able to penetrate even basic armour. Exceptions are creatures that have real life attack power that is as weak as a pillow slap.
Here is the reference:
https://github.com/CombatExtended-Continued/CombatExtended/wiki/Stat-&-Balance-Spreadsheets

**Comment by randomtyper** [comment]
Edit the Patches/Alpha%20Animals/ThingDefs_Misc/Alpha_Hediffs.xml with the lines below to get compatibility for anole and erin hediffs

            <li Class="PatchOperationReplace">
				<xpath>Defs/HediffDef[defName="AA_AnoleGrown"]/stages/li/statOffsets</xpath>
				<value>
                                                                           <statOffsets>
					<ArmorRating_Sharp>3</ArmorRating_Sharp>
					<ArmorRating_Blunt>4.5</ArmorRating_Blunt>
                  			           </statOffsets>
				</value>
			</li>
			
            <li Class="PatchOperationReplace">
				<xpath>Defs/HediffDef[defName="AA_AnoleGrown"]/comps/li[Class="HediffCompProperties_VerbGiver"]/tools/li</xpath>
				<value>
                                                                           <li Class="CombatExtended.ToolCE">  
							<label>left claw</label>
							<capacities>
								<li>Scratch</li>
							</capacities>
							<power>20</power>
							<cooldownTime>2</cooldownTime>
							<linkedBodyPartsGroup>FrontLeftPaw</linkedBodyPartsGroup>
							<surpriseAttack>
								<extraMeleeDamages>
									<li>
										<def>Stun</def>
										<amount>22</amount>
									</li>
								</extraMeleeDamages>
							</surpriseAttack>
							<armorPenetrationSharp>2.5</armorPenetrationSharp>
							<armorPenetrationBlunt>4.5</armorPenetrationBlunt>
						</li>
						<li Class="CombatExtended.ToolCE">
							<label>right claw</label>
							<capacities>
								<li>Scratch</li>
							</capacities>
							<power>20</power>
							<cooldownTime>2</cooldownTime>
							<linkedBodyPartsGroup>FrontRightPaw</linkedBodyPartsGroup>
							<surpriseAttack>
								<extraMeleeDamages>
									<li>
										<def>Stun</def>
										<amount>22</amount>
									</li>
								</extraMeleeDamages>
							</surpriseAttack>
							<armorPenetrationSharp>2</armorPenetrationSharp>
							<armorPenetrationBlunt>4</armorPenetrationBlunt>
						</li>
				</value>
			</li>

**Comment by randomtyper** [comment]
Didn't add any additional attacks for anole in grown state, you can add them if you want.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+				</li>
+				<li Class="PatchOperationReplace">
+					<xpath>/Defs/ThingDef[defName="AA_CrescendoAnole"]/tools</xpath>
+					<value>
+						<tools>
+							<li Class="CombatExtended.ToolCE">
+								<label>left claw</label>
+								<capacities>
+									<li>Scratch</li>
+								</capacities>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml:35]: Anole's offensive capabilities in vanilla in unagitated state are a bit better than these of a domestic cat and a bit worse than that of an arctic fox. This will be a good starting point.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+					</value>
+				</li>
+				<li Class="PatchOperationAdd">
+					<xpath>/Defs/ThingDef[defName="AA_CrescendoAnole"]/statBases</xpath>
+					<!-- No idea for these just did random numbers. -->
+					<!-- (Note this is a lizard that grows strong when angered) -->
+					<value>
+						<MeleeDodgeChance>0.03</MeleeDodgeChance>
+						<MeleeCritChance>0.25</MeleeCritChance>
+						<MeleeParryChance>0.10</MeleeParryChance>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml:24]: These are quite close to statbases of a rough-plated monitor. A good starting point, to be fair. As crescendo anoles are more intelligent, a tad bit higher crit chance and unchanged dodge and parry chance (compared to the monitor, which is much larger and has greater reach and step length) is reasonable. If anything

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrystallineCaracal.xml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+							<bodyShape>Quadruped</bodyShape>
+						</li>
+					</value>
+				</li>
+				<!-- Copied this from the phoenix owl cat -->
+				<li Class="PatchOperationAdd">
+					<xpath>/Defs/ThingDef[defName="AA_CrystallineCaracal"]/statBases</xpath>
+					<value>
+						<MeleeDodgeChance>0.36</MeleeDodgeChance>
+						<MeleeCritChance>0.02</MeleeCritChance>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrystallineCaracal.xml:22]: A crystalline caracal in melee combat is basically a slightly worse lynx. This will be a great starting point.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Erin.xml [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+					</value>
+				</li>
+				<li Class="PatchOperationReplace">
+					<xpath>/Defs/ThingDef[defName="AA_Erin"]/tools</xpath>
+					<value>
+						<tools>
+							<li Class="CombatExtended.ToolCE">
+								<label>left claw</label>
+								<capacities>
+									<li>Scratch</li>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Erin.xml:35]: In vanilla, an erin in melee is essentially a slower, harder hitting warg. Use this as an yet another starting point.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_GreyCoatedMouflon.xml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+				</li>
+				<li Class="PatchOperationReplace">
+					<xpath>/Defs/ThingDef[defName="AA_GreyCoatedMouflon"]/tools</xpath>
+					<value>
+						<tools>
+							<li Class="CombatExtended.ToolCE">
+								<label>left hoof</label>
+								<capacities>
+									<li>Poke</li>
+								</capacities>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_GreyCoatedMouflon.xml:34]: Slightly worse devil sheep. That's all I can say about grey-coated mouflon's combat capabilities. Do not forget about its wool (it is basically a muffalo wool in terms of defensive properties).

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Thunderbeast.xml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+					<xpath>/Defs/ThingDef[defName="AA_Thunderbeast"]/statBases</xpath>
+					<value>
+						<MeleeDodgeChance>0.20</MeleeDodgeChance>
+						<MeleeCritChance>0.15</MeleeCritChance>
+						<MeleeParryChance>0.05</MeleeParryChance>
+					</value>
+				</li>
+				<li Class="PatchOperationReplace">
+					<xpath>/Defs/ThingDef[defName="AA_Thunderbeast"]/tools</xpath>
+					<value>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Thunderbeast.xml:28]: Hard one. Should have ray hound armor, rhino statbases and attack style (with account for its smaller body size) and an electric horn attack. Also, impressive damage output for its body size.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Overgrown_Colossus.xml [hunk 8]
```diff
@@ -2,65 +2,67 @@
 <Patch>
 	<Operation Class="PatchOperationFindMod">
 		<mods>
-		<li>Alpha Animals</li>
+			<li>Alpha Animals</li>
 		</mods>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Overgrown_Colossus.xml:6]: Maybe move all colossi patches into one file to optimize things? They got same stats anyway.

File: Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Projectiles.xml [hunk 9]
```diff
@@ -61,302 +59,301 @@
...
+				</li>
+				<li Class="PatchOperationReplace">
+					<xpath>/Defs/VFECore.ExpandableProjectileDef[defName="AA_AcidicVomit"]/projectile</xpath>
+					<value>
+						<projectile Class="CombatExtended.ProjectilePropertiesCE">
+							<damageDef>AA_AcidSpit</damageDef>
+							<flyOverhead>false</flyOverhead>
+							<damageAmountBase>6</damageAmountBase>
+							<speed>20</speed>
+							<armorPenetrationBlunt>3</armorPenetrationBlunt>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Projectiles.xml:283]: <damageAmountBase>12</damageAmountBase>
							<speed>8</speed>
							<armorPenetrationBlunt>1.5</armorPenetrationBlunt>

I would use these values, as an acidic spit is a more concentrated projectile, it should be faster and better-penetrating than just an unconcentrated stream of acid.

**Comment by randomtyper** [comment]
There is an issue with crystalline caracal: it uses another new projectile. This projectile burns the target and ignites flammable objects on its path (like trees and grass). Here is the code:

```
<VFECore.ExpandableProjectileDef ParentName="BaseBullet">
<defName>AA_LaserBullet</defName>
<label>laser</label>
<thingClass>VFECore.FlamethrowProjectile</thingClass>
<graphicData>
<texPath>Things/Projectiles/Proj_Laser</texPath>
<texPathFadeOut>Things/Projectiles/Proj_Laser</texPathFadeOut>
<shaderType>MoteGlow</shaderType>
</graphicData>
<drawOffscreen>true</drawOffscreen>
<lifeTimeDuration>20</lifeTimeDuration>
<widthScaleFactor>0.75</widthScaleFactor>
<heightScaleFactor>1</heightScaleFactor>
<startingPositionOffset>(0, 0, -1)</startingPositionOffset>
<totalSizeScale>1.5</totalSizeScale>
<tickFrameRate>10</tickFrameRate>
<finalTickFrameRate>5</finalTickFrameRate>
<tickDamageRate>20</tickDamageRate>
<dealsDamageOnce>true</dealsDamageOnce>
<minDistanceToAffect>2</minDistanceToAffect>
<projectile>
<damageDef>Flame</damageDef>
<flyOverhead>false</flyOverhead>
<damageAmountBase>10</damageAmountBase>
<stoppingPower>1.5</stoppingPower>
<ai_IsIncendiary>true</ai_IsIncendiary>
<speed>20</speed>
</projectile>
</VFECore.ExpandableProjectileDef>
```

File: Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml [hunk 10]
```diff
@@ -1,28 +1,49 @@
...
+								<label>claws</label>
+								<capacities>
+									<li>Scratch</li>
+								</capacities>
+								<power>32</power>
+								<cooldownTime>2</cooldownTime>
+								<soundMeleeHit>MeleeHit_BionicPunch</soundMeleeHit>
+								<soundMeleeMiss>MeleeMiss_BionicPunch</soundMeleeMiss>
+								<armorPenetrationSharp>2</armorPenetrationSharp>
+								<armorPenetrationBlunt>3.5</armorPenetrationBlunt>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml]: That's not how CE works AFAIK. You need to specify attack power and penetration for each usable appendage. Also, vanilla attack set is pretty lackluster for a giant intelligent lizard. I used an adult atispec as basis (a very strong creature), and in vanilla version of the mod a grown anole is even better in terms of offensive capabilities (if you only count the claws in).

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml [hunk 11]
```diff
@@ -33,22 +33,38 @@
...
 								<cooldownTime>2.5</cooldownTime>
 								<linkedBodyPartsGroup>FrontLeftClaws</linkedBodyPartsGroup>
-								<armorPenetrationSharp>0.3</armorPenetrationSharp>
-								<armorPenetrationBlunt>1</armorPenetrationBlunt>
+								<surpriseAttack>
+									<extraMeleeDamages>
+										<li>
+											<def>Stun</def>
+											<amount>20</amount>
+										</li>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml:44]: Why did you move the stun here? 20 amount is bonkers even for a rough-plated monitor!

File: Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Projectiles.xml [hunk 12]
```diff
@@ -366,6 +366,18 @@
...
+					<xpath>/Defs/VFECore.ExpandableProjectileDef[defName="AA_LaserBullet"]/projectile</xpath>
+					<value>
+						<projectile Class="CombatExtended.ProjectilePropertiesCE">
+							<damageDef>Flame</damageDef>
+							<flyOverhead>false</flyOverhead>
+							<damageAmountBase>10</damageAmountBase>
+							<ai_IsIncendiary>true</ai_IsIncendiary>
+							<speed>20</speed>
+						</projectile>
+					</value>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Projectiles.xml:380]: Dunno if that will work, it is not that simple. It uses special code from vanilla expanded framework.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml [hunk 13]
```diff
@@ -20,7 +20,7 @@
 					<!-- (Note this is a lizard that grows strong when angered) -->
 					<value>
 						<MeleeDodgeChance>0.03</MeleeDodgeChance>
-						<MeleeCritChance>0.25</MeleeCritChance>
+						<MeleeCritChance>0.29</MeleeCritChance>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml:23]: That change was unnecessary, I just said in the previous comment that your original values were reasonable. But this is still good. Most good predators have better crit chance.

File: Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml (hunk 10 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml]: The Hediff you did for the anole resulted in an error for the patch. I could do the success part for the sequence but idk if that will work.

**Comment by N7Huntsman** [comment]
There is currently a formatting error in the Alpha_Hediffs.xml file. This is do to the patch in there for the "AA_AnoleGrown" having an extra `/` on the end of it.

**Comment by feeddanoob** [comment]
What is the consequence of removing Class=HediffCompProperties_VerbGiver or @HediffCompProperties_VerbGiver from the xpath?

**Comment by feeddanoob** [comment]
Gameplay wise no errors were produced with my changes for the anole but idk if the functionality is the same because of the missing class in the xpath

**Comment by randomtyper** [comment]
> Gameplay wise no errors were produced with my changes for the anole but idk if the functionality is the same because of the missing class in the xpath

Does. Not. Work. I tried a bunch of variations, but anole has the same melee attacks after getting a hediff as before getting a hediff. Something is off.

**Comment by randomtyper** [comment]
Crystalline caracal ranged attack worked as intended, you can readd that patch. The only problem is it being inaccurate as heck. That is relatively easy to fix, though.

**Comment by N7Huntsman** [comment]
I'm not longer getting any start-up errors, the mantrap projectile works, and at a cursory glance I'm not seeing any animals missing stats. Everything seems to be pretty well done at a glance and seems to be working, but I'll do some more substantial playtesting when I have time this weekend.

The only thing I'll remark on is that you might check on any new crafting materials (such as leathers) added for these new creatures. The armor stats may not necessarily need anything done with them, but I just don't see anything here touching them, so they're probably worth glancing at.

Are there outstanding items you're still working on, or do you consider the patch complete at this point?

**Comment by randomtyper** [comment]
> 
> Are there any big, outstanding items you're still working on, or do you consider the patch mostly complete at this point?

@N7Huntsman it is not ready for implementation now, from my point of view. We are having problems with that one anole hediff, that changes its attacks in vanilla version of mod. Also, most values are not final, the new wool is not patched yet (it is just a clone of muffalo wool defence-wise, though), and crystalline caracal ranged attack needs polishing.

@feeddanoob can you confirm or deny my evaluation of readiness?

**Comment by randomtyper** [comment]
Also, we need to patch bodydefs.

**Comment by feeddanoob** [comment]
@randomtyper I didn't know there was a wool to be patched. Also check my latest commit for the anole maybe replacing the whole comps section will work. The body def didn't change (at least not on the body's folder).

**Comment by feeddanoob** [comment]
At least no red errors on startup.

**Comment by randomtyper** [comment]
> @randomtyper I didn't know there was a wool to be patched. Also check my latest commit for the anole maybe replacing the whole comps section will work. The body def didn't change (at least not on the body's folder).

BodyDef comment was a false alarm. Sorry.

File: Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml [hunk 14]
```diff
@@ -1,28 +1,85 @@
...
+												<armorPenetrationBlunt>3.5</armorPenetrationBlunt>
+											</li>
+										</tools>
+									</li>
+									<li Class="HediffCompProperties_Immunizable">
+										<severityPerDayNotImmune>-2</severityPerDayNotImmune>
+									</li>
+									<li Class="AlphaBehavioursAndEvents.HediffCompProperties_AnoleGrown">
+				
+									</li>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml:67]: Lines 62-67 are redundant.

Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml:67]: Even if we do a patchoperationreplace? Idk how rimworld patching works in terms of replacing.

Comment (randomtyper) [reply Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml:67]: These lines replace things with same things. Why add them and bloat the code?

**Comment by randomtyper** [comment]
> 
> 
> At least no red errors on startup.

"If there are no errors detected in code, then compiler has a bug"

**Comment by feeddanoob** [comment]
¯\_(ツ)_/¯

File: Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml (hunk 14 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml:67]: I think since this "works" (theoretical works) why change it? Because all of the comps will be replaced with the lines and wouldn't that fuck with the anole?

**Comment by feeddanoob** [comment]
Mouflan wool is done. I think all the additions are done so I will do the tweaks now.

File: Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml (hunk 14 above)
Comment (randomtyper) [reply Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml:67]: This does not work. The absence of error message can mean that patch did nothing.

File: Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Resources_Leathers.xml [hunk 15]
```diff
@@ -2,313 +2,274 @@
...
-				<nomatch Class="PatchOperationAdd">
-				<xpath>/Defs/ThingDef[defName="Leather_Night"]/statBases</xpath>
+					</value>
+				</li>
+				<!-- SkySteel -->
+				<li Class="PatchOperationAdd">
+					<xpath>/Defs/ThingDef[defName="AA_SkySteel"]/statBases</xpath>
 					<value>
-						<StuffPower_Armor_Sharp>0.08</StuffPower_Armor_Sharp>
+						<Bulk>0.025</Bulk>
```
Comment (randomtyper) [inline Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Resources_Leathers.xml:34]: Dude, can you edit files cleaner? You managed to change 539 lines and possibly f*cked up a few values just to add some stats to wool.

Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Resources_Leathers.xml:34]: I Refactored the xml lines while adding the mouflon wool at the end, nothing was fucked up doing so.

Comment (randomtyper) [reply Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Resources_Leathers.xml:34]: Then sorry. But I'd advice you to refactor files before the changes as a separate commit. Keeps things cleaner.

Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Resources_Leathers.xml:34]: Yeah I should have done that...

**Comment by feeddanoob** [comment]
As for the anole I just saw what you meant with the growth attack and normal attack. Yeah Idk what to do for that in the xml

**Comment by randomtyper** [comment]
> 
> 
> As for the anole I just saw what you meant with the growth attack and normal attack. Yeah Idk what to do for that in the xml file.

*tactical_facepalm.jpg*

**Comment by feeddanoob** [comment]
I would only see the scratch explosive growth before today not the regular anole one....

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml (hunk 2 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml:35]: Done(?)

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml (hunk 3 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml]: Done(?)

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml (hunk 13 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml]: Done(?)

File: Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml (hunk 14 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Misc/Alpha_Hediffs.xml]: Still no idea how to resolve this on the xml side of things...

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrystallineCaracal.xml (hunk 4 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrystallineCaracal.xml]: Done(?)

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Overgrown_Colossus.xml (hunk 8 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Overgrown_Colossus.xml:6]: Mycoid Colossus doesn't have the same stats as an overgrown colossus, so separate files probably needed

File: Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Projectiles.xml (hunk 9 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Misc/AlphaAnimals_CE_Patch_Projectiles.xml]: Done.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml (hunk 11 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_CrescendoAnole.xml]: Done(?)

**Comment by feeddanoob** [comment]
> 
> 
> Crystalline caracal ranged attack worked as intended, you can readd that patch. The only problem is that it is inaccurate as heck. That is relatively easy to fix, though.

Don't know the values I should use for this. I don't know how CE does projectiles =/

**Comment by Breadbox2k19** [comment]
Let me know if you’re having any issues, might be able to help, did all the previous AA patches.

**Comment by Breadbox2k19** [comment]
> > Crystalline caracal ranged attack worked as intended, you can readd that patch. The only problem is that it is inaccurate as heck. That is relatively easy to fix, though.
> 
> Don't know the values I should use for this. I don't know how CE does projectiles =/

It’s a bit hard to explain this in words, how about you ignore this issue for now? And I fix this in a subsequent update?

**Comment by randomtyper** [comment]
> Let me know if you’re having any issues, might be able to help, did all the previous AA patches.

We are having an issue with the anole hediff. That hediff replaces anole's attacks and we can't make it work. I hope you know what I'm talking about. 
Also, as you are the maker of previous AA patches, what do you think increasing rasorjack's statbases? I find them quite low for a predator.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Erin.xml (hunk 5 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Erin.xml:35]: Done(?)

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_GreyCoatedMouflon.xml (hunk 6 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_GreyCoatedMouflon.xml:34]: Done(?)

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Thunderbeast.xml (hunk 7 above)
Comment (feeddanoob) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Thunderbeast.xml:28]: Done(?)

**Comment by feeddanoob** [comment]
> 
> 
> > > Crystalline caracal ranged attack worked as intended, you can readd that patch. The only problem is that it is inaccurate as heck. That is relatively easy to fix, though.
> > 
> > 
> > Don't know the values I should use for this. I don't know how CE does projectiles =/
> 
> It’s a bit hard to explain this in words, how about you ignore this issue for now? And I fix this in a subsequent update?

Yeah I figured... btw Thank you for creating the alpha animals patch. I think I am done with the edits on this PR so someone else can contribute if needed. So far the Hediff and projectiles are something I don't know how to balance/fix.

**Review by Breadbox2k19 — APPROVED** [review]
Will finalise details at a later patch.

**Review by N7Huntsman — APPROVED** [review]
